### PR TITLE
Added a helper function to make putting empty objects easier

### PR DESCRIPTION
**Pull request by chanchett**

### CODE REVIEW

File: test/test.cpp [hunk 1]
```diff
@@ -72,9 +72,15 @@ void clear_bucket(const ds3_client* client, const char* bucket_name) {
     handle_error(error);
 
     for (i = 0; i < bucket_response->num_objects; i++) {
-        request = ds3_init_delete_object(bucket_name, bucket_response->objects[i].name->value);
-        error = ds3_delete_object(client, request);
-        ds3_free_request(request);
+        if (bucket_response->objects[i].name->value[bucket_response->objects[i].name->size-1]!='/') {
```
Comment (chanchett) [inline test/test.cpp]: I wasn't sure if this was actually a good way to check if the object is a folder or not. Is there a better way of determining that?

Comment (DenverM80) [inline test/test.cpp]: seems reasonable to me.  The trailing slash is the only distinguishing property of a folder.

**Comment by rpmoore** [comment]
Why would you want to create a list of empty objects?

**Comment by chanchett** [comment]
So that you can make an empty folder on the server. I could rename the function to make that more clear?

**Comment by rpmoore** [comment]
It seems like we should have a call that allows them to create a single directory, rather than multiple.  Creating directories is not a common thing, and is not something that we really recommend.

**Comment by chanchett** [comment]
Turns out that the problem with adding an empty "folder" was just that the trailing slash was being stripped, which was fixed in another pull request. So this one is now just a unit test to make sure that it still works.

**Pull request merged**